subiquity/server/controllers/install.py

```diff
@@ -130,12 +160,97 @@ async def configure_apt(self, *, context):
         configurer = await mirror.wait_config()
         return await configurer.configure_for_install(context)
 
+    async def run_curtin_install_step(
+            self, step: CurtinInstallStep,
+            resume_data_file: Path,
+            context, source) -> subprocess.CompletedProcess:
+        """ Run a curtin install step. """
+
+        self.app.note_file_for_apport(
+                f"Curtin{step.name}Config", str(step.config_file))
+        self.app.note_file_for_apport(
+                f"Curtin{step.name}Errors", str(step.error_file))
+        self.app.note_file_for_apport(
+                f"Curtin{step.name}Log", str(step.log_file))
+
+        self.write_config(
+                config_file=step.config_file,
+                config=step.acquire_config(step, resume_data_file)
+                )
+
+        # Make sure the log directory exists.
+        step.log_file.parent.mkdir(parents=True, exist_ok=True)
+
+        return await run_curtin_command(
+                self.app, context,
+                "install", source,
+                "--set", f'json:stages={json.dumps(step.stages)}',
+                config=str(step.config_file),
+                private_mounts=False)
+
     @with_context(
         description="installing system", level="INFO", childlevel="DEBUG")
     async def curtin_install(self, *, context, source):
-        await run_curtin_command(self.app, context, 'install', source,
-                                 config=self.write_config(),
-                                 private_mounts=False)
+        # Until https://bugs.launchpad.net/curtin/+bug/1876984 gets
+        # fixed, the only way to get curtin to leave the network
+        # config entirely alone is to omit the 'network' stage.
+
+        logs_dir_prefix = Path(
```

Comment:
In general I would support using Path more in the subiquity codebase btw. There are a few things I don't really like about it but it's better than paths as strings...

Reply:
I agree. I don't love pathlib but I don't hate it either.

The thing that frustrates me the most is how difficult it is to "concatenate" two absolute paths without "losing" the first one . The following is pretty common in Subiquity:
```python
path1 = Path("/target")
path2 = Path("/var/log/installer")

path1 / path2
# Gives PosixPath('/var/log/installer') but we want PosixPath('/target/var/log/installer')
```

. To be fair, `os.path.join` suffers from the same thing but I was hoping `pathlib` would provide a simpler way to deal with that.